### ISSUE TITLE
Crash app when back to previous page

### ISSUE THREAD

**Issue by thangdev** [Yoctol/react-messenger-customer-chat#73]
System Information
OS: Ubuntu 20.04 LTS
Browser: Chrome Version 89.0.4389.82

Context
Plugin was implemented in Reactjs v16.13.1
I set the appId="" (blank string) and the module still works well

Steps to replicate
When the page is loaded, the messenger work well but when I click go back button on browser to navigate to previous page of my web, it crashes my web

![image](https://user-images.githubusercontent.com/30391934/112414458-a7fbd300-8d54-11eb-804f-a7cd53d4aa78.png)

**Comment by AnLuubk** [comment]
while wait for feedback you can try: https://levelup.gitconnected.com/integrating-facebook-customer-chat-into-a-react-js-app-5b7c21343048

**Comment by thangdev** [comment]
> while wait for feedback you can try: https://levelup.gitconnected.com/integrating-facebook-customer-chat-into-a-react-js-app-5b7c21343048 :))

Hi @AnLuubk , Did you get the error like me ?

**Comment by AnLuubk** [comment]
i got both 2 error you just created today and i just hotfix by the link i just comment previous. May be facebook have new update that package works incorrectly :(

**Comment by thangdev** [comment]
@AnLuubk , Hi, thank you so much, I am follow this solution you mentioned above.
But I am stuck at `import { ConfigContext } from '../hooks';`
I don't understand this line. What is the ConfigContext ? and what is inside hooks folder ?
Could you please send me the detail example your code ?
Thank you so much !!!

**Comment by AnLuubk** [comment]
you dont need to do the same just create component like this:

`import React from 'react';
import { fb } from '../../utils/thirdPartyAPi';


export default class MessengerCustomerChat extends React.PureComponent {
  componentDidMount() {
    this.timeout = setTimeout(() => {
      fb(FB => this.timeout && FB.XFBML.parse());
    }, 2000);
  }

  componentWillUnmount() {
    clearTimeout(this.timeout);
    delete this.timeout;
  }

  render() {
    return <div className="fb-customerchat" attribution="setup_tool" page_id="1950359978511589"/>
  }
}`

**Comment by khaphannm** [comment]
> @AnLuubk , Hi, thank you so much, I am follow this solution you mentioned above.
> But I am stuck at `import { ConfigContext } from '../hooks';`
> I don't understand this line. What is the ConfigContext ? and what is inside hooks folder ?
> Could you please send me the detail example your code ?
> Thank you so much !!!

The ConfigContext just think it is a place to get your PageID and AppID constant, he manage those information using React context

**Comment by AnLuubk** [comment]
> > @AnLuubk , Hi, thank you so much, I am follow this solution you mentioned above.
> > But I am stuck at `import { ConfigContext } from '../hooks';`
> > I don't understand this line. What is the ConfigContext ? and what is inside hooks folder ?
> > Could you please send me the detail example your code ?
> > Thank you so much !!!
> 
> The ConfigContext just think it is a place to get your PageID and AppID constant, he manage those information using React context

yes! for hiding data :)

**Comment by thangdev** [comment]
Hi, @AnLuubk , @khaphannm I got the error like this: 
![image](https://user-images.githubusercontent.com/30391934/112461652-bb2d9380-8d92-11eb-8ca3-18e99ea0dfd1.png)
I think the reason is **window.config.facebook** is null, how can I fix this ?
And do we really need an facebook appId ? Thank you! 
![image](https://user-images.githubusercontent.com/30391934/112462107-4444ca80-8d93-11eb-8905-2e87b870f88f.png)

**Comment by AnLuubk** [comment]
> Hi, @AnLuubk , @khaphannm I am implementing it but now I got an error like this:
> ![image](https://user-images.githubusercontent.com/30391934/112461652-bb2d9380-8d92-11eb-8ca3-18e99ea0dfd1.png)
> I think the reason is **window.config.facebook** is null, how can I fix this ?
> And do we really need a facebook appId variable here ? Thank you!
> ![image](https://user-images.githubusercontent.com/30391934/112462107-4444ca80-8d93-11eb-8905-2e87b870f88f.png)

you just put appId to direct value or empty string: "". facebook caht plugin does not need appId (i think)

**Comment by khaphannm** [comment]
Hi @thangdev
I feel the approach in the link is not quite convenient, so you can copy the source file as yours
https://github.com/Yoctol/react-messenger-customer-chat/blob/master/src/MessengerCustomerChat.js

Then just modify a bit like this:
`window.fbAsyncInit = () => {
      window.FB.init({
        appId,
        autoLogAppEvents,
        xfbml,
        version: `v${version}`,
      });

      this.setState({ fbLoaded: true });
      // Add this line
      window.FB.XFBML.parse();
 };`

**Comment by thangdev** [comment]
Hi @khaphannm, 
Thanks for your support. Yesterday,  I tried set the appId to empty string as @AnLuubk said and it works  well now :))

**Comment by thangdev** [comment]
@AnLuubk The only problem now is the messenger doesn't hide when I click to navigate to other page

**Comment by khaphannm** [comment]
Yes, you can choose which approach is easier for you, because this library is more convenient for controlling the props, when you need to custom your plugin.

You can have a look this example, my component, and everything is just in one place, to set up your own plugin
```
<MessengerCustomerChat
           pageId={process.env.GATSBY_PAGE_ID}
           appId={process.env.GATSBY_APP_ID}
           version={process.env.GATSBY_MESSENGER_PLUGIN_VERSION}
           loggedInGreeting={process.env.GATSBY_MESSENGER_MSG_IN}
           loggedOutGreeting={process.env.GATSBY_MESSENGER_MSG_OUT}
           htmlRef={window.location.pathname}
           themeColor={secondaryMain}
           language="vi_VN"
           greetingDialogDisplay="show"
           greetingDialogDelay={parseInt(process.env.GATSBY_MESSENGER_PLUGIN_DELAY)}
           shouldShowDialog
         />
```

**Comment by AnLuubk** [comment]
> @AnLuubk The only problem now is the messenger doesn't hide when I click to navigate to other page, it's still there.
> I load this component only at specific child component but It still appears in all pages

you can check the pathname if this is the route you want to render Messenger

**Comment by AnLuubk** [comment]
Or you can try method show and hide like this: https://stackoverflow.com/a/64451277

**Comment by AnLuubk** [comment]
> Yes, you can choose which approach is easier for you, because this library is more convenient for controlling the props, when you need to custom your plugin.
> 
> You can have a look this example, my component, and everything is just in one place, to set up your own plugin
> 
> ```
> <MessengerCustomerChat
>            pageId={process.env.GATSBY_PAGE_ID}
>            appId={process.env.GATSBY_APP_ID}
>            version={process.env.GATSBY_MESSENGER_PLUGIN_VERSION}
>            loggedInGreeting={process.env.GATSBY_MESSENGER_MSG_IN}
>            loggedOutGreeting={process.env.GATSBY_MESSENGER_MSG_OUT}
>            htmlRef={window.location.pathname}
>            themeColor={secondaryMain}
>            language="vi_VN"
>            greetingDialogDisplay="show"
>            greetingDialogDelay={parseInt(process.env.GATSBY_MESSENGER_PLUGIN_DELAY)}
>            shouldShowDialog
>          />
> ```

this is better code but im not sure how the code deal with server render, it may crash before come to client. if it work can you explain more?? :)

**Comment by khaphannm** [comment]
> > Yes, you can choose which approach is easier for you, because this library is more convenient for controlling the props, when you need to custom your plugin.
> > You can have a look this example, my component, and everything is just in one place, to set up your own plugin
> > ```
> > <MessengerCustomerChat
> >            pageId={process.env.GATSBY_PAGE_ID}
> >            appId={process.env.GATSBY_APP_ID}
> >            version={process.env.GATSBY_MESSENGER_PLUGIN_VERSION}
> >            loggedInGreeting={process.env.GATSBY_MESSENGER_MSG_IN}
> >            loggedOutGreeting={process.env.GATSBY_MESSENGER_MSG_OUT}
> >            htmlRef={window.location.pathname}
> >            themeColor={secondaryMain}
> >            language="vi_VN"
> >            greetingDialogDisplay="show"
> >            greetingDialogDelay={parseInt(process.env.GATSBY_MESSENGER_PLUGIN_DELAY)}
> >            shouldShowDialog
> >          />
> > ```
> 
> this is better code but im not sure how the code deal with server render, it may crash before come to client. if it work can you explain more?? :)

Server side render will return all needed content for u when receiving request, then cannot crash

**Comment by AnLuubk** [comment]
sure i think i still have to check if it is client render for script load to load. Thank you for your help. I will refactor my code later!

**Comment by khaphannm** [comment]
@AnLuubk @thangdev 
Now i got a problem. The chat only appear when I logged in facebook already, otherwise it will not. Do u guys experience that error?
